app/src/main/java/me/hufman/androidautoidrive/carapp/notifications/views/DetailsView.kt

```diff
@@ -229,10 +231,28 @@ class DetailsView(val state: RHMIState, val phoneAppResources: PhoneAppResources
 			clearButton.setEnabled(false)
 		}
 
+		val hasReadoutAction = !notificationSettings.shouldReadoutNotificationDetails()
+		val readoutButton = buttons[1]
+		if (hasReadoutAction) {
+			readoutButton.apply {
+				setEnabled(true)
+				getTooltipModel()?.asRaDataModel()?.value = L.NOTIFICATION_READOUT_ACTION
+				getAction()?.asRAAction()?.rhmiActionCallback = RHMIActionButtonCallback {
+					getAction()?.asHMIAction()?.getTargetModel()?.asRaIntModel()?.value = state.id
+					readoutInteractions.triggerReadout(notification)
+				}
+			}
+		} else {
+			readoutButton.apply {
```

Comment:
For this branch, where the readoutButton isn't shown, I think the following `for` loop will configure this `buttons[1]` with the normal action logic, and so this branch doesn't need to do anything.

Reply:
Agree, the subsequent loop shall dynamically configure all buttons that are not assigned to a hardcoded action (clear or readout) before.

Reply:
refactored